test-cypress/integration/explore/graphs.overview.spec.js

```diff
@@ -1,114 +1,131 @@
+    }
+
+    function confirmDelete() {
+        cy.get('.modal-footer .confirm-btn').click();
+        cy.get('.modal').should('not.be.visible');
+    }
+
+    context('Test graphs overview pagination', () => {
+        it('Should be visible', () => {
+            cy.get('div[paginations]')
+                .should('be.visible')
+                .and('contain', '3');
+            verifyGraph('The default graph');
+        });
+
+        it('Should switch pages', () => {
+            // Switch through pages and verify that the respective pager button is active.
+            selectPage(2).should('contain', '2')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:11');
+
+            selectPage(3).should('contain', '3')
+                .closest('li').should('be', 'active');
+            verifyGraph('urn:21');
+        });
     });
 
-    it('Test graphs overview search', function () {
-        cy.wait(1000);
-        cy.get('.form-control').type(graphToSearch);
-        cy.verifyGraphExists(graphToSearch);
-        cy.get("tbody").should('not.contain', "urn:11");
+    it('Test items per page switch menu', () => {
+        // Change number of items per page from the menu and verify that proper rows are present in
+        // the table.
+        selectItemFromMenu(1);
+        verifyGraph('urn:19');
 
+        selectItemFromMenu(2);
+        verifyGraph('urn:1');
+        verifyGraph('urn:21');
     });
 
-    it('Delete graph', function () {
-        cy.wait(1000);
-        cy.get('tbody tr a')
-            .each(($el, index) => {
-                if ($el.attr('title') === graphToDelete) {
-                    cy.get('[class="fa fa-trash-o fa-lg"]').eq(index).click({force: true});
-                    cy.get('.modal-content button.btn.btn-primary').click({force: true});
-                }
-            });
-        cy.get("tbody").should('not.contain', graphToDelete);
+    it('Test graphs overview search', () => {
+        // Default items/rows per page is 10.
+        verifyVisibleGraphsCountIs(10);
+        // Type graph name in the filter field and verify that proper graphs remain visible in the
+        // table.
+        cy.get('.search-graphs').type('urn:2');
+        verifyVisibleGraphsCountIs(3);
+        verifyGraph('urn:2');
+        verifyGraph('urn:11', false);
     });
 
-    it('Clear repository - should delete all graphs', function () {
-        cy.get(".clearfix button.btn.btn-secondary").contains('Clear').click();
-        cy.get(".modal-content button.btn.btn-primary").click();
-        cy.get("tbody").should('not.contain', "urn");
-        cy.verifyGraphExists("The default graph");
-
+    it('Delete graph', () => {
+        let graphToDelete = 'urn:9';
+        cy.get(`${EXPORT_GRAPHS_TABLE_ID} tbody a[title="${graphToDelete}"]`)
+            .closest('tr').find('.delete-graph-btn').click();
+        confirmDelete();
+        verifyGraph(graphToDelete, false);
```

Comment:
Well I forgot what `verifyGraph` does exactly and it's not clear what is verified here. Probably another method `verifyGraphIsMissing`  or rename to `verifyGraphExistence` ?

Reply:
You are right, this name is not very descriptive. I'll use the `verifyGraphExistence`